sentry-core/src/main/java/io/sentry/core/PiiEventProcessor.java

```diff
@@ -0,0 +1,33 @@
+package io.sentry.core;
+
+import io.sentry.core.protocol.User;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Removes personal identifiable information from {@link SentryEvent} if {@link
+ * SentryOptions#isSendDefaultPii()} is set to false.
+ */
+@ApiStatus.Internal
+final class PiiEventProcessor implements EventProcessor {
+  private final @NotNull SentryOptions options;
+
+  PiiEventProcessor(final @NotNull SentryOptions options) {
+    this.options = options;
```

Comment:
we could use `Objects.requireNonNull` just for the sake of not getting NPE on `options.isSendDefaultPii()`

Reply:
just reading it again and as this is internal, package-private, it's not really necessary, we are the only one creating it anyway.